WordPress/src/main/java/org/wordpress/android/ui/FilteredRecyclerView.java

```diff
@@ -526,6 +608,89 @@ public boolean isFirstItemVisible() {
         return (child != null && mRecyclerView.getLayoutManager().getPosition(child) == 0);
     }
 
+    class FilteredPagerAdapter extends PagerAdapter {
+        private RecyclerView mRecyclerView;
+        private List<FilterCriteria> mFilterCriteria;
+
+        FilteredPagerAdapter(RecyclerView recyclerView, List<FilterCriteria> filterCriteria) {
+            mRecyclerView = recyclerView;
+            mFilterCriteria = filterCriteria;
+        }
+
+
+        private boolean datasetChanged(List<FilterCriteria> filterCriteriaOptions) {
+            boolean isChanged = false;
+
+            if (mFilterCriteria.size() != filterCriteriaOptions.size()) {
+                isChanged = true;
+            } else {
+                for (int index = 0; index < mFilterCriteria.size(); index++) {
+                    if (!mFilterCriteria.get(index).equals(filterCriteriaOptions.get(index))) {
+                        isChanged = true;
+                        break;
+                    }
+                }
+            }
+
+            return isChanged;
+        }
+
+        public void updateFiltersIfNeeded(List<FilterCriteria> filterCriteriaOptions) {
+            if (datasetChanged(filterCriteriaOptions)) {
+                mFilterCriteria = filterCriteriaOptions;
+                mSelectingRememberedFilterOnCreate = true;
+                notifyDataSetChanged();
+            }
+        }
+
+        public FilterCriteria getFilterAtPosition(int position) {
+            return mFilterCriteria.get(position);
+        }
+
+        @Override public int getItemPosition(@NonNull Object object) {
+            if (object != null && mFilterCriteria != null) {
+                for (int i = 0; i < mFilterCriteria.size(); i++) {
+                    FilterCriteria criteria = mFilterCriteria.get(i);
+                    if (criteria != null && criteria.equals(object)) {
+                        return i;
+                    }
+                }
+            }
+            return -1;
+        }
+
+        @Override public int getCount() {
+            return mFilterCriteria.size();
+        }
+
+        @Override public boolean isViewFromObject(@NonNull View view, @NonNull Object object) {
+            return view == object;
+        }
+
+        @Nullable @Override public CharSequence getPageTitle(int position) {
```

Comment:
I think the annotations should be all on separate lines all over this PR. I'm surprised we don't have that in checkstyle 👍

Reply:
You are right sorry. I tried to change at least the obvious one 😊.